'All' Pages all done + bug fixes

R-Eli-Levasseur opened the pull request:
- Added duplicate entry checks and feedback for add admin
- Removed isActive, non-specific gender,  not-provided, and distance filter
- Made delete in allUsers work, deletes user and all their events, then rebuilds index
- Hooked up allEvents page and set delete event to rebuild index
- Added security to all 'all' pages
- Added number of events to all providers
- Fixed bug where providers could not delete their events from provider.php
- Fixed bug where changing name in provider would cause events to not show up until reload
- Fixed bug where cancelling an event edit would cause a page break
- Fixed bug where fees would appear to 1 decimal point instead of 2

R-Eli-Leva committed:
'All' Pages all done + bug fixes

-Added duplicate entry checks and feedback for add admin
- Removed isActive, non-specific gender,  not-provided, and distance filter
- Made delete in allUsers work, deletes user and all their events, then rebuilds index
-Hooked up allEvents page and set delete event to rebuild index
-Added security to all 'all' pages
-Added number of events to all providers
- Fixed bug where providers could not delete their events from provider.php
- Fixed bug where changing name in provider would cause events to not show up until reload
- Fixed bug where cancelling an event edit would cause a page break
- Fixed bug where fees would appear to 1 decimal point instead of 2

justwheel commented on grasa_event_locator/forms.py:
```diff
@@ -28,7 +28,7 @@
     ("Other", "Other"),
 ]
-transportationList = [("Provided", "Provided"), ("Not-Provided", "Not-Provided")]
+transportationList = [("Provided", "Provided")]
```
This change was made because the filters are now opt-in, i.e. you can choose to filter by events that provide transportation by checking it, or you can not filter. Is this correct?

justwheel commented on grasa_event_locator/models.py:
```diff
@@ -6,7 +6,6 @@ class userInfo(models.Model):
     user = models.OneToOneField(User, on_delete=models.CASCADE)
     org_name = models.CharField(max_length=255)
     isAdmin = models.BooleanField(default=False)
-    isActive = models.BooleanField(default=False)
```
This will require a database schema change, right? Do we need to commit a migration file?

R-Eli-Levasseur replied:
Yes it requires a schema change, committing migration files has not worked well though, should drop db and rerun makemigrations/migrate

justwheel commented on grasa_event_locator/templates/allEvents.php:
```diff
@@ -40,14 +41,14 @@
                       <div class="modal-footer">
                         <button type="button" class="btn btn-secondary" data-dismiss="modal">Never Mind</button>
-                        <button type="button" class="btn btn-danger" data-dismiss="modal" onclick="">Confirm Delete</button>
+                        <button type="button" class="btn btn-danger" data-dismiss="modal" onclick="window.location.href='{% url 'deny_event' p.id %}'">Confirm Delete</button>
```
This button deletes the event, not just deny it, right? I was wondering about the method name.

leong96 replied:
@jwflory From my quick read of the code for denyEvent, it's a delete (drops the entry from the table).

R-Eli-Levasseur replied:
Yes it deletes the program, might want to refactor to a specific delete function but they would both do the same thing, only point I would see would be to have a different email sent

justwheel commented on grasa_event_locator/templates/event.php:
```diff
@@ -24,7 +24,7 @@
               <ul class="list-group list-group-flush">
                 {% if event.fees %}
-                <li class="list-group-item"><i class="fa fa-money fa-fw" aria-hidden="true"></i> ${{ event.fees }}</li>
+                <li class="list-group-item"><i class="fa fa-money fa-fw" aria-hidden="true"></i> ${{ fees }}</li>
```
Oh, huh. Was this all required to fix #136?

R-Eli-Levasseur replied:
No, I had to put the decimal logic in views.py because jijna didn't like it

justwheel commented on grasa_event_locator/templates/header.php:
```diff
@@ -44,17 +44,17 @@
 
         //Distance
-        var distanceList = ["0-5 miles", "6-10 miles", "11-15 miles", "16-20 miles", "20+ miles"]
+        //var distanceList = ["0-5 miles", "6-10 miles", "11-15 miles", "16-20 miles", "20+ miles"]
```
Why was this line commented out?

R-Eli-Levasseur replied:
We decided in the last back-end meeting that we were dropping the distance filter, I probably should have just deleted it

justwheel commented on grasa_event_locator/urls.py:
```diff
@@ -8,7 +8,6 @@
 urlpatterns = [
     path('aboutContact.php', views.aboutContact),
     path('admin.php', views.admin, name='admin_page'),
-    path('admin_activate', views.admin_activate),
```
Is this the URL to create an initial admin user?

leong96 replied:
@jwflory That was the url that would reactivate the admin account if it was accidentally disabled while we were testing the delete user modal functionality. The button was removed from the header, so it's associated URL is no longer needed.

R-Eli-Levasseur replied:
Exactly as Harrison said, it was no longer needed so I removed it

leong96 commented on grasa_event_locator/views.py:
```diff
@@ -163,30 +164,34 @@ def event(request, eventID):
 
-        timing_list = event.categories.filter(id__gte=29)
-        timing_list = timing_list.filter(id__lte=34)
+        timing_list = event.categories.filter(id__gte=27)
```
Very good, you noticed that this part needed to be changed.

R-Eli-Levasseur replied:
Only after it broke my tests haha

leong96 reviewed (approved):
Looks swell, though I guess answer Justin's comments first before merging?

leong96 committed:
Remove Django User

This commit adds on to the delete user feature; as it will now also delete the Django User, in addition to removing the database entry.

Additionally, admin users are now also made superusers to log into the django admin portal. If a person were to create a superuser via createsuperuer, it would mess up the ID numbering in the table.

nlMeminger commented:
@eguy006 please resolve the conflict here

Commit:
Merge branch 'master' into all-pages
